data_analysis_pipeline/word2vec_timeline_utils.py

```diff
@@ -0,0 +1,248 @@
+                "_".join(model_two[1].stem.split("_")[0:2]),
+                output_filepath.parent,
+            )
+            for model_one, model_two in itertools.product(
+                zip(year_one_models, year_one_model_paths),
+                zip(year_two_models, year_two_model_paths),
+            )
+        ]
+
+    else:
+        return [
+            (
+                model_one[0],
+                model_two[0],
+                "_".join(model_one[1].stem.split("_")[0:2]),
+                "_".join(model_two[1].stem.split("_")[0:2]),
+                output_filepath.parent,
+            )
+            for model_one, model_two in itertools.combinations(
+                zip(year_one_models, year_one_model_paths), r=2
+            )
+        ]
+
+
+def calculate_distances(argument_batch):
+    """
+    This function is designed to calculate distances for each model pair.
+    It also writes the output to a file.
+
+    Argument:
+        argument_batch - a tuple that contains arguments for running this function (designed for parallel processing)
+    """
+
+    model_one = argument_batch[0]
+    model_two = argument_batch[1]
+    year_one = argument_batch[2]
+    year_two = argument_batch[3]
+    parent_folder = argument_batch[4]
+
+    temp_filename = (
+        f"{str(parent_folder)}/calculated_{year_one}-{year_two}_intra_distance_temp.tsv"
+    )
+
+    common_vocab = set(model_one.wv.vocab.keys()) & set(model_two.wv.vocab.keys())
+
+    with open(temp_filename, "w") as outfile:
+        writer = csv.DictWriter(
+            outfile, delimiter="\t", fieldnames=["tok", "year_pair", "distance"]
+        )
+        writer.writeheader()
+        for tok in tqdm.tqdm(common_vocab):
+            writer.writerow(
+                dict(
+                    tok=tok,
+                    year_pair=f"{year_one}-{year_two}",
+                    distance=cdist([model_one.wv[tok]], [model_two.wv[tok]], "cosine")[
+                        0
+                    ][0],
+                )
+            )
+
+    return temp_filename
+
+
+def merge_calculated_distances(temp_filepaths, outfile_path):
+    with open(outfile_path, "w") as outfile:
+        writer = csv.DictWriter(
+            outfile, delimiter="\t", fieldnames=["tok", "year_pair", "distance"]
+        )
+        writer.writeheader()
+        for filename in temp_filepaths:
+            with open(filename, "r") as infile:
+                reader = csv.DictReader(
+                    infile, delimiter="\t", fieldnames=["tok", "year_pair", "distance"]
+                )
+                for entry in tqdm.tqdm(reader):
+                    writer.writerow(entry)
+            Path(filename).unlink(missing_ok=True)
+
+
+def train_word2vec(doc_iterator):
+    sentence_iterator = doc_iterator[0]
+    output_folder = doc_iterator[1]
+    seed_idx = doc_iterator[2]
+    model_category = doc_iterator[3]
+
+    seed = 100 + seed_idx
```

Comment:
Why do you start your seeds at 100? Is it because the default seed for Word2Vec is 100 and used elsewhere or something? Or is it just personal preference?

Reply:
Personal preference, but I can allow users to put in whatever seed they want. Will change to allow that feature.